### PR TITLE
Ft request notification 168662102

### PR DESCRIPTION
**Pull request by OlawaleJoseph**
#### What does this PR do?
 Enables users to receive notifications when they create trip request

#### Description of Task to be completed?
  - Ensure staff's manager receives email notification when a direct report creates travel request.
  - Ensure users can opt-out of email notifications.
  - Ensure users receive web notification.

#### How should this be manually tested?
  - git clone this repo using the command `git clone git@github.com:andela/firestar-backend.git`
  - cd to the project
  - checkout to the branch using the command `git checkout ft-create-trips-167730747`
  - create a `.env` file and set PORT,JWT_SECRET and DB variables as specified in the .env.example file.
  - Run the command `npm install` in the terminal.
  - Open a terminal and run `npm test`.
  - Open a terminal and run `npm run migrate:dev` and `npm run seed:dev`.
  - Open your browser(preferably chrome or firefox), visit `http://localhost:<PORT>`.
  1. Open Postman, make a post request to `localhost:<port>/api/v1/users/auth/login`, with any of the below 
            in the request body.
- `{email: requester1@gmail.com, password: Password1$}`
2. Copy the token in the reponse
3. Make a POST request to `http:localhost:<port>/api/v1/requests` with the token obtained above in the request header as `authorization`=`Bearer <token obtained>`
 4. Send a request object containing the following for a one-way trip:
         `{
           reason: 'Business',
           departmentId: 2,
           tripType: 'oneWay',
        }`
 
 6. Make a POST request to `http:localhost:<port>/api/v1/requests/<requestId>/trips` with the token obtained previously set the the authorization header.
     i. Send a request object containing the following to create a trip request
          `{
            trip: {
                destinationLocationId: 2,
                departureLocationId: 1,
                accommodationId: 2, 
                departureDate: '2020/10/10 18:00'
              }
         }`
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#168662053](https://www.pivotaltracker.com/story/show/168662102)

#### Screenshots (if appropriate)
Screenshot (97)

Screenshot (98)

#### Questions:
N/A

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -19,9 +19,11 @@ SOME_PASSWORD="" (description: password to create a test account, must include n
 ADMIN_EMAIL = email to use as superadmin in seeding data to the database
 ADMIN_PASSWORD =  password to use as superadmin password in seeding data to the database.
 JWT_SECRET = any random secret key
+PRIVATE_VAPID_KEY = private key generated from web-push module
```
Comment (zanio) [inline .env.example]: How do i have this key

File: .env.example [hunk 2]
```diff
@@ -19,9 +19,11 @@ SOME_PASSWORD="" (description: password to create a test account, must include n
...
+PUBLIC_VAPID_KEY = public key generated from web-push module
```
Comment (zanio) [inline .env.example]: How do i have this key value to ?

File: src/routes/api/notifications.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ *        description: success
+ *        content:
+ *          application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/response'
+ */
+router.post('/notifications/notify', async (req, res) => {
+  const { subscription, message } = req.body;
+  res.status(201).json({
+    success: true
```
Comment (zanio) [inline src/routes/api/notifications.js]: can you abstract this logic into it own controller in the controller folder

Comment (zanio) [inline src/routes/api/notifications.js]: can you abstract this logic into it own controller in the controller folder

File: .env.example (hunk 1 above)
Comment (OlawaleJoseph) [reply .env.example]: PRIVATE_VAPID_KEY = -JSFK2xhiTWcy31v0fnsajHrhNNqFfyhsKhcBVfCpfY
PUBLIC_VAPID_KEY = BDBWrChIwSY82qo-sN2Oi7tzOA5nhyXVonLm312OPKIVhw1CAjhT0l9FIVJcifRHQzvC0nBEI2nK4CWjSHt6mb8

File: src/routes/api/notifications.js (hunk 3 above)
Comment (OlawaleJoseph) [reply src/routes/api/notifications.js]: Will do

Comment (OlawaleJoseph) [reply src/routes/api/notifications.js]: Will do

**Comment by zanio** [comment]
Hello bro, sorry for the delay. But i did not get any notification when i successfully create a request ?

**Comment by OlawaleJoseph** [comment]
> Hello bro, sorry for the delay. But i did not get any notification when i successfully create a request ?

The manager gets the notification not the requester

**Comment by zanio** [comment]
> > Hello bro, sorry for the delay. But i did not get any notification when i successfully create a request ?
> 
> The manager gets the notification not the requester

So what do i have to do to play the role of a manager and a requester ?

**Comment by zanio** [comment]
Please i am back let's get to work again

**Comment by OlawaleJoseph** [comment]
> Please i am back let's get to work again

Good to hear. I didn't see this earlier.
Set the env variable MANAGER_EMAIL to your email

**Comment by zanio** [comment]
> > Please i am back let's get to work again
> 
> Good to hear. I didn't see this earlier.
> Set the env variable MANAGER_EMAIL to your email

okay i would do that

**Pull request merged**